libnmstate/validator.py

```diff
@@ -357,6 +359,40 @@ def _assert_vlan_filtering_trunk_tag(trunk_tag_state):
             )
 
 
+def _assert_vlan_mtu_not_greater_than_base_iface(
+    original_desired_state, current_state
+):
+    # validate if vlan mtu is greater than base interface mtu
+    for iface_name, iface_state in original_desired_state.interfaces.items():
+        if iface_state.get(Interface.TYPE) == InterfaceType.VLAN:
+            if iface_state.get(Interface.STATE) == InterfaceState.UP:
+                base_iface_name = iface_state.get(VLAN.CONFIG_SUBTREE).get(
+                    VLAN.BASE_IFACE
+                )
+                # check if base_iface is in desired state.
+                # if not then get from current state
+                base_iface_state = original_desired_state.interfaces.get(
+                    base_iface_name
+                )
+                if not base_iface_state:
+                    base_iface_state = current_state.interfaces.get(
+                        base_iface_name
+                    )
+                if not base_iface_state:
+                    raise ValueError(
+                        f"base interface {base_iface_name} for vlan \
+                        {iface_name} does not exist"
+                    )
+
+                base_iface_mtu = base_iface_state.get(Interface.MTU)
+                vlan_mtu = iface_state.get(Interface.MTU)
+                if vlan_mtu and base_iface_mtu and (vlan_mtu > base_iface_mtu):
```

Comment:
I think `vlan_mtu` and `base_iface_mtu` cannot be `None` at this point. I suggest you to get them directly with `[]` and there is no need to check them here. Thanks!

Reply:
In one (or maybe more) test cases they are coming as None
`test_create_vlan_over_a_bond_slave` is one of the tests failing if I don't check that.

Full past failure log: https://pastebin.com/Jf4nVPNB 
from https://travis-ci.com/github/nmstate/nmstate/jobs/336141093

Reply:
Fair enough, thanks!